internal/flags/apiflags/flags.go

```diff
@@ -0,0 +1,152 @@
+// Copyright 2020 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package apiflags
+
+import (
+	"fmt"
+	"strings"
+
+	gencrd "github.com/operator-framework/operator-sdk/internal/generate/crd"
+	"github.com/operator-framework/operator-sdk/internal/util/projutil"
+	"github.com/spf13/pflag"
+)
+
+// AddAPIFlags ...
+type AddAPIFlags struct { //nolint:golint
+
+	APIVersion       string
+	Kind             string
+	OperatorType     string
+	SkipGeneration   bool
+	Repo             string
+	GeneratePlaybook bool
+	HeaderFile       string
+	GitInit          bool
+	MakeVendor       bool
+	SkipValidation   bool
+	CrdVersion       string
+
+	HelmChartRef     string
+	HelmChartVersion string
+	HelmChartRepo    string
+}
+
+// AddTo - Add the reconcile period and watches file flags to the the flagset
+// helpTextPrefix will allow you add a prefix to default help text. Joined by a space.
+func (f *AddAPIFlags) AddTo(flagSet *pflag.FlagSet) {
+	flagSet.StringVar(&f.APIVersion,
+		"api-version",
+		"",
+		"Kubernetes apiVersion and has"+
+			" a format of $GROUP_NAME/$VERSION (e.g app.example.com/v1alpha1) - used with \"ansible\" or \"helm\" types")
+	flagSet.StringVar(&f.Kind,
+		"kind",
+		"",
+		"Kubernetes resource Kind name. (e.g AppService)")
+	flagSet.StringVar(&f.OperatorType,
+		"type",
+		"go",
+		"Type of operator to initialize (choices: \"go\", \"ansible\" or \"helm\")")
+	flagSet.StringVar(&f.Repo,
+		"repo",
+		"",
+		"Project repository path for Go operators. Used as the project's Go import path. This must be set if "+
+			"outside of $GOPATH/src (e.g. github.com/example-inc/my-operator)")
+	flagSet.BoolVar(&f.GitInit,
+		"git-init",
+		false,
+		"Initialize the project directory as a git repository (default false)")
+	flagSet.StringVar(&f.HeaderFile,
+		"header-file",
+		"",
+		"Path to file containing headers for generated Go files. Copied to hack/boilerplate.go.txt")
+	flagSet.BoolVar(&f.MakeVendor,
+		"vendor",
+		false,
+		"Use a vendor directory for dependencies")
+	flagSet.BoolVar(&f.SkipValidation,
+		"skip-validation",
+		false,
+		"Do not validate the resulting project's structure and dependencies. (Only used for --type go)")
+	flagSet.BoolVar(&f.GeneratePlaybook, "generate-playbook", false,
+		"Generate a playbook skeleton. (Only used for --type ansible)")
+
+	flagSet.BoolVar(&f.SkipGeneration,
```

Comment:
@joelanford would not it also be removed when we remove the openapi option?

Reply:
No, because we do `generate k8s` and `generate crds`.